plugin/pkg/scheduler/algorithm/priorities/priorities.go

```diff
@@ -170,6 +170,77 @@ func (n *NodeLabelPrioritizer) CalculateNodeLabelPriority(pod *api.Pod, machines
 	return result, nil
 }
 
+// This is a reasonable size range of all container images. 90%ile of images on dockerhub drops into this range.
+const (
+	mb         uint64 = 1024 * 1024
+	minImgSize uint64 = 23 * mb
+	maxImgSize uint64 = 1000 * mb
+)
+
+// ImageLocalityPriority is a priority function that favors nodes that already have requested pod container's images.
+// It will detect whether the requested images are present on a node, and then calculate a score ranging from 0 to 10
+// based on the total size of those images.
+// - If all the images are not present on a node, this node will be given the lowest priority.
+// - If some of the images are present on a node, the larger their sizes' sum, the higher the node's priority.
+func ImageLocalityPriority(pod *api.Pod, machinesToPods map[string][]*api.Pod, podLister algorithm.PodLister, nodeLister algorithm.NodeLister) (schedulerapi.HostPriorityList, error) {
+	sumSizeMap := make(map[string]uint64)
+
+	nodes, err := nodeLister.List()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, container := range pod.Spec.Containers {
+		for _, node := range nodes.Items {
+			// Check if this container's image is present and get its size.
+			imageSize := checkContainerImageOnNode(node, container)
+			// Add this size to the total result of this node.
+			sumSizeMap[node.Name] += imageSize
+		}
+	}
+
+	result := []schedulerapi.HostPriority{}
+	// score int - scale of 0-10
+	// 0 being the lowest priority and 10 being the highest.
+	for nodeName, sumSize := range sumSizeMap {
+		var score int
+		switch {
+		case sumSize == 0 || sumSize < minImgSize:
```

Comment:
The code would be a little simpler if you move this logic into calculateScoreFromSize(). So this would look like

```
for nodeName, sumSize := range sumSizeMap {
    result = append(result, schedulerapi.HostPriority{Host: nodeName, Score: claculateScoreFromSize(sumSize)})
}
return result, nil
```

Then claculateScoreFromSize() would be like

```
switch {
  case sumSize == 0 || sumSize < minImgSize:
     score = 0
  case sumSize >= maxImgSize:
      score = 10
  default:
      score = int((10 * (sumSize - minImgSize) / (maxImgSize - minImgSize)) + 1)
```